great_expectations/data_context/store/checkpoint_store.py

```diff
@@ -177,22 +177,76 @@ def get_checkpoint(
+
+        Args:
+            checkpoint: The Checkpoint to persist.
+
+        Returns:
+            The persisted Checkpoint (possibly modified state based on store backend).
+
+        Raises:
+            CheckpointError: If a Checkpoint with the given name already exists.
+        """
         key = self._construct_key_from_checkpoint(checkpoint)
-        return self._add_checkpoint(key=key, checkpoint=checkpoint)
+
+        try:
+            checkpoint_ref = self.add(key=key, value=checkpoint.get_config())  # type: ignore[func-returns-value]
+        except gx_exceptions.StoreBackendError:
+            raise gx_exceptions.CheckpointError(
+                f"A Checkpoint named {checkpoint.name} already exists."
+            )
+
+        if isinstance(checkpoint_ref, GXCloudIDAwareRef):
+            cloud_id = checkpoint_ref.cloud_id
+            checkpoint.ge_cloud_id = uuid.UUID(cloud_id)  # type: ignore[misc]
+
+        return checkpoint
 
     def update_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint:
+        """Use a stand-alone Checkpoint object to update a persisted value.
+
+        Args:
+            checkpoint: The Checkpoint to use for updating.
+
+        Returns:
+            The persisted Checkpoint (possibly modified state based on store backend).
+
+        Raises:
+            CheckpointNotFoundError: If a Checkpoint with the given name does not exist in the store.
+        """
         key = self._construct_key_from_checkpoint(checkpoint)
-        if not self.has_key(key):
+
+        try:
+            checkpoint_ref = self.update(key=key, value=checkpoint.get_config())  # type: ignore[func-returns-value]
+        except gx_exceptions.StoreBackendError:
             raise gx_exceptions.CheckpointNotFoundError(
-                f"Could not find a Checkpoint named {checkpoint.name}"
+                f"Could not find an existing Checkpoint named {checkpoint.name}."
             )
-        return self._add_checkpoint(key=key, checkpoint=checkpoint)
+
+        if isinstance(checkpoint_ref, GXCloudIDAwareRef):
+            cloud_id = checkpoint_ref.cloud_id
+            checkpoint.ge_cloud_id = uuid.UUID(cloud_id)  # type: ignore[misc]
+
+        return checkpoint
 
     def add_or_update_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint:
+        """Use a stand-alone Checkpoint object to either add or update a persisted value.
+
+        Args:
+            checkpoint: The Checkpoint to use for adding or updating.
+
+        Returns:
+            The persisted Checkpoint (possibly modified state based on store backend).
+        """
         key = self._construct_key_from_checkpoint(checkpoint)
-        if self.has_key(key):
-            return self.update_checkpoint(checkpoint)
-        return self._add_checkpoint(key=key, checkpoint=checkpoint)
+
+        checkpoint_ref = self.add_or_update(key=key, value=checkpoint.get_config())  # type: ignore[func-returns-value]
```

Comment:
If this is actually returning something shouldn't we just update the return type on the `add_or_update` method?
https://mypy.readthedocs.io/en/stable/error_code_list.html?highlight=func-returns-value#check-that-called-function-returns-a-value-func-returns-value

Reply:
ah yeah you are right. This is bad design here - normally return `None` but in the Cloud case, we return this ref. I'm wary of adding that annotation to the base `Store` so where might this go?